packages/blueprint-component-mapper/src/files/select.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import FormGroupWrapper from './form-group';
+
+import { Select as BSelect, MultiSelect } from '@blueprintjs/select';
+import propsCatcher from '../common/props-catcher';
+import { MenuItem, Button } from '@blueprintjs/core';
+
+export const itemRenderer = (item, { handleClick }, value) => (
+  <MenuItem
+    key={item.value}
+    {...item}
+    {...(!item.text && { label: '', text: item.label })}
+    onClick={handleClick}
+    icon={item.value === value || (Array.isArray(value) && value.includes(item.value)) ? 'tick' : 'blank'}
+  />
+);
+
+export const multiOnChange = (item, input) =>
+  input.value && !input.value.includes(item.value)
+    ? input.onChange([...input.value, item.value])
+    : input.onChange(input.value ? input.value.filter((v) => v !== item.value) : [item.value]);
+
+export const tagRenderer = (item, options) => {
+  const option = options.find(({ value }) => value === item);
+
+  if (option) {
+    return option.text || option.label;
+  }
+};
+
+export const itemPredicate = (query, item) => {
+  if (!query) {
+    return item;
+  }
+
+  if (item.label.toLowerCase().includes(query.toLowerCase())) {
+    return item;
+  }
+};
+
+const Select = ({ input, options, placeholder, isSearchable, noOptionsMessage, isMulti, ...props }) => {
```

Comment:
https://github.com/data-driven-forms/react-forms/pull/477/commits/e35daf51bcd0ed4e70b0e170225f340cb7ee1f6b

Here the react select has been extracted from the common select and it now requires SelectComponent prop. That will allow using all the features (especially async) with the library select. I would make a note here to do that before we release the 2.4. I want to migrate the MUI select aswell.

Reply:
Yeah, I was also thinking about making a hook or something from that functionality.